web/impact/impact/v1/views/base_list_view.py

```diff
@@ -32,18 +39,45 @@ def metadata(self):
         return result
 
     def get(self, request):
-        limit = int(request.GET.get('limit', 10))
-        offset = int(request.GET.get('offset', 0))
+        limit = self._validate_kwarg(
+            request.GET.get("limit", str(self.DEFAULT_LIMIT)), key="limit")
+        offset = self._validate_kwarg(
+            request.GET.get("offset", str(0)), key="offset")
+        if self.errors:
+            return Response(status=401, data=self.errors)
         base_url = _base_url(request)
         count, results = self.results(limit, offset)
         result = {
             "count": count,
-            "next": _url(base_url, limit, offset + limit),
-            "previous": _url(base_url, limit, offset - limit),
+            "next": _next_url(base_url, limit, offset, count),
+            "previous": _previous_url(base_url, limit, offset, count),
             "results": results,
         }
         return Response(result)
 
+    def _validate_kwarg(self, val, key):
+        if not is_int(val):
+            self.errors.append(KWARG_VALUE_NOT_INTEGER_ERROR.format(key))
+            return None
+        if key == "limit":
+            self._validate_limit(key, val)
+        if key == "offset":
+            self._validate_offset(key, val)
+        return int(val)
+
+    def _validate_offset(self, key, val):
+        if int(val) < 0:
+            self.errors.append(
+                KWARG_VALUE_IS_NON_POSITIVE_ERROR.format(key))
```

Comment:
Technically 0 is not a positive integer and it seems like it would be better to have two error messages since the two places this is used are different.  I would say something that doesn't hinge on the user understanding exactly what you mean by "positive integer" like "value of '{}' should be greater than zero" or "value of '{}" should be greater than or equal to zero".